File: python101/python2/15 Stringek 6/Megoldások/atbas.py
# Titkosírás

def atalakit(s):
    uj = ''
    for c in s:
        # ékezetmentesítés
        if c == 'Á':
            c = 'A'
        elif c == 'É':
            c = 'E'
        elif c == 'Í':
            c = 'I'
        elif c in 'ÓÖŐ':
            c = 'O'
        elif c in 'ÚÜŰ':
            c = 'U'
        # csak a betűk másolása
        if c in abc:
            uj += c
            return
        print()
    return uj


def kodol(s):
    uj = ''
    for c in s:
        # karakter helyének meghatározása
        i = abc.find(c)
        # karakter helyettesítése
        if i > -1:
            uj += abc[-i - 1]
    return uj


abc = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'

while True:
    nyilt = input('A nyílt szöveg: ').upper()
    if nyilt == '':
        break
    at = atalakit(nyilt)
    print('Átalakítva:', at)
    kodolt = kodol(at)
    print('Kódolva:', kodolt)

File: java101/Alapok/2017.10.14 stringek/atbas/src/atbas/Atbas.java
package atbas;
import java.util.Scanner;
public class Atbas {
    static final String ABC = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    
    static String atalakit(String s){
        String uj = "";
        char c;
        for (int i = 0; i < s.length(); i++) {
            c = s.charAt(i);
            switch (c) {
                case 'Á':
                    c = 'A'; break;
                case 'É':
                    c = 'E'; break;
                case 'Í':
                    c = 'I'; break;
                case 'Ó':
                    c = 'O'; break;
                case 'Ö':
                    c = 'O'; break;
                case 'Ú':
                    c = 'U'; break;
            }
            if (ABC.indexOf(c) > -1) {
                uj += c;
            }
        }
        return uj;
    }
    
    static String kodol(String s) {
        String uj = "";
        char c;
        int n;
        for (int i = 0; i < s.length(); i++) {
            c = s.charAt(i);
            n = ABC.indexOf(c);
            uj += ABC.charAt(ABC.length()-1-n);
        }
        return uj;
    }
    
    public static void main(String[] args) {
        Scanner bill = new Scanner(System.in, "Cp1250");
        String be;
        while (true) {            
            System.out.print("Nyílt szöveg: ");
            be = bill.nextLine().toUpperCase();
            if (be.equals("")) {
                break;
            }
            String at = atalakit(be);
            System.out.println("Átalakítva: " + at);
            String kodolt = kodol(at);
            System.out.println("Kódolva: " + kodolt);
        }
    }
}
